_posts/2024-12-19-cosmicqc.md

```diff
@@ -0,0 +1,285 @@
+
+In conclusion, I want to thank the authors of this paper for their hard work and their important contributions to pushing the field towards developing quality control methodology at the single-cell level for morphology profiles.
+This paper fueled the fire that I had started in myself about wanting to pursue working on a new methodology, one that is simple, to filter out poor quality segmentations from single-cell morphology profiles.
+
+---
+
+## Introduction to coSMicQC
+
+After 2 and half years of being in my position and performing image-based profiling, I have come to realize that I am not perfect (shocking I know! 😱).
+No matter how much I try to manually optimize my segmentation parameters in CellProfiler or when using [Cellpose](https://www.cellpose.org/) or [Stardist](https://stardist.net/), it will not work for all of the single cells in my datasets.
+Now, I am not saying this is a fault of my own, but segmentation errors are a persistent issue and is currently a problem that hasn't been solved to this day.
+
+My point is, you can expect in all of your datasets that it will contain some poorly segmented cells that can be described as technical artifacts.
+You can also expect that debris or smudges could also end up being mis-segmented as cells.
+When working with single-cell data, we do not want these technical artifacts impacting the interpretation of our results.
+Thus, was born this simple idea in my head.
+
+Why not just filter out single cells based on the extracted morphology features?
+
+You wouldn't need to train any model or perform any other pipeline.
+You already have the data in your lap to use, but it is how you use that data is important.
+
+In comes, [coSMicQC](https://github.com/WayScience/coSMicQC), or Single cell Morphology Quality Control!
+
+This new method works by creating conditions, represented by dictionaries.
+A condition is a feature or group of features that will detect a specific type of technical artifact.
+For example, nuclei area and intensity can work together as a condition to detect large nuclei with high intensity which represents segmentation of clustered/overlapping nuclei.
+You condition will also need to include the standard deviation from the mean in which you expect these outliers to be located.
+If you want to detect cells above a mean, you use a positive value, and vice versa for cells below the mean.
+
+{%
+  include figure.html
+  image="images/blog/cosmicqc/area_outlier_example.png"
+  width="100%"
+%}
+> Example of how outliers are detected based on nuclei area in coSMicQC.
+
+We then apply the function `find_outliers` to the simple conditions independently, yielding a dataframe with only the identified outliers, including metadata and the features used to detect the outliers.
```

Comment:
```suggestion
Then apply the function `find_outliers` to the simple conditions and voila, coSMicQC yields a dataframe with flagged single-cell outliers.
```

True? What you have is a bit confusing to me

Reply:
One other thing that was a bit unclear to me (putting my naive reader hat on 🎩 ) is "apply" - what do you mean apply, how do I do this? I think to prevent this question, you need to add a paragraph describing that it is open-source software that we developed with sustainable software development practices. You can even shoutout to @d33bs 📣

Reply:
100% agreed! I will update